File: UVA/10684_-_The_jackpot/main.cpp
#include <cstdio>
#define max(a, b) a >= b ? a : b
using namespace std;

int main() {
  int N;
  while (true) {
    scanf("%d\n", &N);
    if (N == 0)
      break;
    int x[N], s[N], sum = 0, max_sum = 0, min_sum = 0;
    for (int i = 0; i < N; i++) {
      scanf("%d", &x[i]);
    }
    for (int i = 0; i < N; i++) {
      sum += x[i];
      if (sum < min_sum)
        min_sum = sum;
      if (sum - min_sum > max_sum) {
        max_sum = sum - min_sum;
      }
    }
    if (max_sum == 0)
      printf("Losing streak.\n");
    else
      printf("The maximum winning streak is %d.\n", max_sum);
  }
}

File: UVA/10684_-_The_jackpot/Main.java
import java.io.*;
import java.util.*;

public class Main {
  public static PrintWriter out;
  public static int[] nums;
  public static void main(String[] args) {
    MyScanner sc = new MyScanner();
    out = new PrintWriter(new BufferedOutputStream(System.out));
    while (true) {
      int n = sc.nextInt();
      if (n == 0)
        break;
      nums = new int[n];
      for (int i = 0; i < n; i++) {
        nums[i] = sc.nextInt();
      }
      
      int streak = 0;
      int sum = 0, min_sum = 0;
      for (int i = 0; i < n; i++) {
        sum += nums[i];
        min_sum = Math.min(sum, min_sum);
        streak = Math.max(sum - min_sum, streak);
      }

      if (streak > 0)
        out.printf("The maximum winning streak is %d.\n", streak);
      else
        out.println("Losing streak.");

    }
  
    out.close();
  }


  public static class MyScanner {
    BufferedReader br;
    StringTokenizer st;

    public MyScanner() {
       br = new BufferedReader(new InputStreamReader(System.in));
    }
 
    public boolean hasNext() {
      try {
        boolean a = br.ready();
        return a;
      } catch (IOException e) {
        return false;
      }
    }

    public String next() {
      while (st == null || !st.hasMoreElements()) {
        try {
          st = new StringTokenizer(br.readLine());
        } catch (IOException e) {
          e.printStackTrace();
        }
      }
      return st.nextToken();
    }
 
    public int nextInt() {
      return Integer.parseInt(next());
    }
 
    public long nextLong() {
      return Long.parseLong(next());
    }

    public double nextDouble() {
      return Double.parseDouble(next());
    }

    public char nextChar() {
      return next().charAt(0);
    }

    public String nextLine() {
      String str = "";
      try {
         str = br.readLine();
      } catch (IOException e) {
         e.printStackTrace();
      }
      return str;
    }
  }
}
